frontend/src/utils/TimeWithOffset.js

```diff
@@ -34,6 +36,18 @@ class TimeWithOffset {
   toString () {
     return `${this.hours}:${this.minutes} GMT${this.offsetSign}${this.offsetHours}:${this.offsetMinutes}`
   }
+
+  getNextTimestampISOString () {
+    let nextTimestamp = moment(`${this.hours}${this.minutes}${this.offsetSign}${this.offsetHours}${this.offsetMinutes}`, 'HHmmZ')
+    if (!nextTimestamp.isValid()) {
+      return
+    }
+    if (nextTimestamp.isBefore(moment())) {
+      nextTimestamp = nextTimestamp.add(1, 'd')
+    }
+
+    return nextTimestamp.toISOString()
+  }
```

Comment:
```suggestion

  nextDate () {
    let date = new Date()
    const now = date.getTime()
    date = new Date(date.toISOString().substring(0, 11) + this.getTimeString() + this.getTimezoneString())
    while (date.getTime() <= now) {
      date.setUTCDate(date.getUTCDate() + 1)
    }
    return date
  }
```

Reply:
The method returns the next date in the future with the same time as given by the object.

Reply:
It is not sufficient to add 1 day only once!